Add new information methods (`isActiveRaw`, `isEnabledRaw` and `show`)

deancsmith opened the pull request:
This PR adds three new methods to help get additional information about services. We are aiming to use this library to display information about several services on a web dashboard, with the potential management of these being a secondary concern.

To this end, I've added some additional info methods:

- `isActiveRaw()`: retrieve the raw string output of the `is-active` command. We need this as we use some 'oneshot' services, which can be 'inactive' (as opposed to failed) under normal circumstances. Simply checking them to see if they are 'active', as the normal `isActive()` method does, is not enough for us: we need to raw response from the `systemctl` command. 
- `isEnabledRaw()`: retrieve the raw string output of the `is-active` command. We don't strictly need the output of this but I added it for completeness.
- `show()`: retrieve the raw string output of the `show` command but parsing it to an array. We need some more detailed information like the time a 'oneshot' last ran, and potentially the `ExecStart` line to see the command actually run for debugging. This method takes the (extensive) output of the `show` command and makes it available to be easily consumed via a correctly keyed array of the output of the command.

Happy to make changes to the above to better conform with your library. I just quickly added these methods so we could access this additional systemd information.

deancsmith committed:
Added raw variants of isActive/isEnabled

deancsmith committed:
Add `show` command for additional unit info

deancsmith committed:
Style fixes and updated README

icanhazstring commented:
Hi. Thanks for using the lib and this PR. Will look over it asap. :+1:

icanhazstring commented on src/Unit/AbstractUnit.php:
```diff
@@ -158,6 +157,19 @@ public function isEnabled(): bool
+    public function isEnabledRaw(): string
+    {
+        // We have to trim() the output, as it may end in a newline character that we don't want.
+        $output = \trim($this->execute('is-enabled')->getOutput());
```
Great thanks. One thing tho, can you remap the `isActive` and `isEnabled` method to use the output of `isActiveRaw` and `isEnabledRaw` so we can reduce code duplication.

deancsmith replied:
Good point! Done.

deancsmith committed:
Reuse raw methods within boolean methods

icanhazstring commented:
Thanks! Merged and tagged with `0.7.0`.

icanhazstring commented:
Merged into 1.x and tagged with `1.0-rc2`.
Are you using the current 1.x branch for your usage, or the currenty release `0.6.1`?
Didn't notice you filed the PR against `1.x`

deancsmith commented:
Thanks!

I only started looking at your library about an hour before I created the PR. I tried the `0.6.x` branch but was having some issues/was looking for more info methods, so I switched to the newer `1.x` branch. I still couldn't get the info that I needed, so I added these methods in a fork of the `1.x` branch.

We may just keep using the fork in the short-term, as we are only using this info on a dev/debugging page at the moment. Once you get to `1` then we'll switch across, but don't worry about expediting that or backporting to `0.6.x` on account of us.

FYI we are also using this on RHEL 7, which ships with a pretty old systemd (v219) (but I haven't noticed any negative effects of this).

icanhazstring commented:
> FYI we are also using this on RHEL 7, which ships with a pretty old systemd (v219) (but I haven't noticed any negative effects of this).

The major difference I came upon is the format of `on_calendar` setting for timer.
But since your are not creating a unit with this lib. You won't encounter any problems ;)
